core/src/main/java/io/grpc/transport/ServerTransportListener.java

```diff
@@ -48,4 +49,9 @@
    */
   ServerStreamListener streamCreated(ServerStream stream, String method,
       Metadata.Headers headers);
+
+  /**
+   * The transport completed shutting down. All resources have been released.
+   */
+  void transportTerminated();
```

Comment:
> I think a single state makes the API way more simplified and understandable ... and IMHO that outweighs the cons in this case.

I don't think we are going to make any progress. I don't see how such an API could simpler than two zero-arg methods that return boolean and the users not having to combine them.